### ISSUE TITLE
[dove] Error: schema with key or id "UsersData" already exists on client

### ISSUE THREAD

**Issue by lukvermeulen** [feathersjs/feathers#2739]
### Steps to reproduce

I created an app that imports the client from the backend: 
```typescript
import { createClient } from "api";
import rest from "@feathersjs/rest-client";
import authentication from "@feathersjs/authentication-client";

const apiUrl = "http://localhost:3030";

const fetchConnection =
    typeof window !== "undefined" ? window.fetch.bind(window) : fetch;

const restClient = rest(apiUrl);

const client = createClient(restClient.fetch(fetchConnection));
client.configure(authentication());

export default client;
```

with:
"@feathersjs/authentication-client": "pre",
"@feathersjs/rest-client": "pre",

My client is a NextJS site, that currently just calls `client.service("users").find()`. The client appears to be typed, which is very nice, and in total working. I am not sure if it correlates, but this error definitely appears when the API is not even running.

Edit: 
The issue occurs in the repo as seen here: https://github.com/lukvermeulen/feathers-monorepo-example

This is the error: 
```
error - Error: schema with key or id "UsersData" already exists
    at Ajv._checkUnique (D:\02_GitHub\information-tiles\node_modules\ajv\dist\core.js:465:19)
    at Ajv._addSchema (D:\02_GitHub\information-tiles\node_modules\ajv\dist\core.js:456:22)
    at Ajv.compile (D:\02_GitHub\information-tiles\node_modules\ajv\dist\core.js:158:26)
    at new SchemaWrapper (D:\02_GitHub\information-tiles\node_modules\@feathersjs\schema\lib\schema.js:17:35)
    at schema (D:\02_GitHub\information-tiles\node_modules\@feathersjs\schema\lib\schema.js:43:12)
    at eval (webpack-internal:///../api/lib/services/users/users.schema.js:6:47)
    at Object.../api/lib/services/users/users.schema.js (D:\02_GitHub\information-tiles\apps\web-client\.next\server\pages\index.js:33:1)
    at __webpack_require__ (D:\02_GitHub\information-tiles\apps\web-client\.next\server\webpack-runtime.js:33:43)
    at eval (webpack-internal:///../api/lib/client.js:19:14)
    at Object.../api/lib/client.js (D:\02_GitHub\information-tiles\apps\web-client\.next\server\pages\index.js:22:1) {
  page: '/'
}
```

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):

**NodeJS version**: 16.13.0

**Operating System**: Windows Pro 64 bit

**Comment by lukvermeulen** [comment]
Also adding on this: Is this the best way to define the client with a rest connection now? Testing the page with `@testing-library/react` complains like this: 

```
 FAIL   web-client  apps/web-client/specs/index.spec.tsx
  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'bind')

      11 | const restClient = rest(apiUrl);
      12 |
    > 13 | const client = createClient(restClient.fetch(window.fetch.bind(window)));
         |                                                           ^
      15 |
      16 | export default client;

      at Object.<anonymous> (feathers.ts:13:59)
      at Object.<anonymous> (pages/index.tsx:3:1)
      at Object.<anonymous> (specs/index.spec.tsx:4:1)
      at TestScheduler.scheduleTests (../../node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (../../node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (../../node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (../../node_modules/@jest/core/build/cli/index.js:173:3)
```

**Comment by lukvermeulen** [comment]
The issue with "schema [..] already exists" could be related to #2681 and solved there.

**Comment by daffl** [comment]
That was my thought. Hopefully it will be fixed in the next prerelease.

**Comment by daffl** [comment]
Closing since this should be fixed in v5.0.0-pre.29
